### DISCUSSION TITLE
New Panasonic PIOT-V2(WA) wifi interface spotted in the wild

### DISCUSSION THREAD

**Discussion by techydude** [DomiStyle/esphome-panasonic-ac#116] [General]
I recently bought a Panasonic multi-split RCAC, installed last February 2024 (2x CS-Z25ZKEW inside blowers, & 1x CU-2Z50TBE outside compressor).  I'd read about ESPhome/ESP32-based replacements (based on this repo) for the Panasonic DSNK-P11 or a CZ-TACG1, expecting one of them to be in mine.  Nope!  There's a new cost-reduced wifi interface in town!

Mine has a 'PIOT-V2(WA)' in it, and there appears to be at least one other variant of it 'CA' instead of 'WA'.
(copies of compliance docs here:  https://manuals.plus/panasonic/piot-v2wa-wireless-module-manual )

It looks like the attached photo, just a bare board on a plastic mounting frame, with what might be the same 4-wire (grey) interface as its predecessors, but with I think a different connector.

Anyone else aware of this, & tested this firmware on an ESP32 with this newer Panasonic heat-pump model?
If not, I'll give it a go as time permits...

![IMG_5251](https://github.com/DomiStyle/esphome-panasonic-ac/assets/3955601/ad82dd5d-d621-4e56-9088-94ca049a01ee)

**Comment by host-felice** [comment]
Did you find any other info about it? I am afraid I could have the same controller... And I cannot check it now because they are in another city. I planned to replace them, but I would love not to go there and find a surprice -_-

**Reply by techydude** [reply]
Unfortunately no, I haven't looked into this any further, yet.
But I'll bump it up my list of projects to try current firmware.

**Reply by host-felice** [reply]
I had the possibility to read a Panasonic manual for that AC and it seems that DSNK-P11 and PIOT-V2(WA) are probably the same or at least interchangeable. DSNK looks like an esp8266 while PIOT an esp32 because it has the BLE and WPA3 (it is in italian, but the tech info are readable). The last sentence says that, more or less, you can use one of them for your AC internal unit and to ask to your retailer.

![Senza titolo](https://github.com/user-attachments/assets/b7cd1812-adab-4152-ac6e-937840fa776e)

What do you think?

**Comment by host-felice** [comment]
hey @techydude I found a pic of those Panasonic and guess what? They have the CN-CNT connector too. Check yours! Maybe you can just unplug the piot and go with the other connector! Easy peasy
![Senza titolo](https://github.com/user-attachments/assets/e16c72cf-f06b-4c3c-a5fd-eeacc26206c2)

**Comment by Thibaultbbb** [comment]
Has anyone tried the conversion of the new WIFI modules PIOT-V2(WA)?

**Reply by jonasf21** [reply]
I just finished this successfully converting a CS-Z35ZKEW with the new PIOT-V2 to ESPHOME. As already mentioned, the new module is esp32. I cant see a level shifter (5V->3V) so i assume panasonic doesnt do it. There is alot of discussion about ESP32 GPIO being 5V tolerant without a final answer. I have converted all my panasonic (both DNSK-P11 and PIOT-V2) to ESP32 with ESPHOME without level converter and so far all are fine. 
As for the code: i am getting some messages about shifted RX packets and it appears i have to "reboot" the ESP32 every time when the AC is cut off from power. (I use the remote for that, simply press the wifi button twice with a paper clip etc). I guess panasonic updated some parts of the code but its still backward compatible. However the support for the Ethera gen is not very good (i made a quick and dirty fork) but atleast it works.
As for the PIOT-V2: the module now has a 12-pin connector instead of 8-pin. However the pin layout is the same, you just have to deduct the last 4 pins. See below pictures and comparision between DNSK-P11 vs. PIOT-V2.
![conn-pana](https://github.com/user-attachments/assets/0c052276-8372-4e06-bdd1-2c8fdce17ad0)
![20250621_211709](https://github.com/user-attachments/assets/ac4bf7f9-d684-467d-a5bb-d794a83060d6)
![20250621_211723](https://github.com/user-attachments/assets/a3359977-4f3f-4fda-9cd7-7d99cd1f4c0a)
